substrate/primitives/genesis-builder/src/lib.rs

```diff
@@ -19,39 +19,90 @@
 
 //! Substrate genesis config builder
 //!
-//! This Runtime API allows to construct `RuntimeGenesisConfig`, in particular:
-//! - serialize the runtime default `RuntimeGenesisConfig` struct into json format,
-//! - put the RuntimeGenesisConfig struct into the storage. Internally this operation calls
-//!   `GenesisBuild::build` function for all runtime pallets, which is typically provided by
-//!   pallet's author.
-//! - deserialize the `RuntimeGenesisConfig` from given json blob and put `RuntimeGenesisConfig`
-//!   into the state storage. Allows to build customized configuration.
+//! This module provides means to interact with `RuntimeGenesisConfig`. Runtime provides a default
+//! `RuntimeGenesisConfig` structure in a form of the JSON blob.
 //!
-//! Providing externalities with empty storage and putting `RuntimeGenesisConfig` into storage
-//! allows to catch and build the raw storage of `RuntimeGenesisConfig` which is the foundation for
-//! genesis block.
+//! Additionally the runtime may provide a number of partial predefined `RuntimeGenesisConfig`
+//! configurations in the form of patches which shall be applied on top of the default
+//! `RuntimeGenesisConfig`. The patch is a JSON blob, which essentially comprises the list of
+//! key-value pairs that are to be customized in the default runtime genesis config.
+//! These predefined configurations are referred to as presets.
+//!
+//! This allows the runtime to provide a number of predefined configs (e.g. for different
+//! testnets or development) without neccessity to leak the runtime types outside the itself (e.g.
+//! node or chain-spec related tools).
+//!
+//! This Runtime API allows to interact with `RuntimeGenesisConfig`, in particular:
+//! - provide the list of available preset names,
+//! - provide a number of named presets of `RuntimeGenesisConfig`,
+//! - provide a JSON represention of the default `RuntimeGenesisConfig` (by simply serializing the
+//!   default `RuntimeGenesisConfig` struct into JSON format),
+//! - deserialize the full `RuntimeGenesisConfig` from given JSON blob and put the resulting
+//!   `RuntimeGenesisConfig` structure into the state storage creating the initial runtime's state.
+//!   Allows to build customized genesis. This operation internally calls `GenesisBuild::build`
+//!   function for all runtime pallets.
+//!
+//! Providing externalities with an empty storage and putting `RuntimeGenesisConfig` into storage
+//! (by calling `build_state`) allows to construct the raw storage of `RuntimeGenesisConfig`
+//! which is the foundation for genesis block.
 
 extern crate alloc;
+use alloc::vec::Vec;
+use codec::{Decode, Encode};
+use scale_info::TypeInfo;
 
 /// The result type alias, used in build methods. `Err` contains formatted error message.
 pub type Result = core::result::Result<(), sp_runtime::RuntimeString>;
 
+/// Wrapper representing preset id.
+#[derive(Decode, Encode, Debug, TypeInfo, PartialEq)]
+pub struct PresetId(Vec<u8>);
```

Comment:
I think this is fine, if you want to be pedantic you can use something like `BoundedVec<u8>` to have a limit on the size, or use a fixed size byte array as identifier.

Reply:
This is not on-chain so I see no point in imposing size limits here. Why would you use `BoundedVec` here?

Reply:
done in: 71589d9ef0a